core/src/main/java/tc/oc/pgm/menu/items/InventoryItem.java

```diff
@@ -0,0 +1,51 @@
+package tc.oc.pgm.menu.items;
+
+import org.bukkit.event.inventory.ClickType;
+import org.bukkit.inventory.ItemStack;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.menu.InventoryMenu;
+
+/** Interface that describes a selection in an inventory menu. */
+public interface InventoryItem {
+
+  /**
+   * Generates an item stack from the inventory item which is in a specific inventory for a specific
+   * player. This method is called to generate an {@link ItemStack} to fill {@link
+   * org.bukkit.inventory.Inventory} with for the inventory gui api.
+   *
+   * <p>This method can return null, in which case the item should not be rendered and is treated as
+   * the player not having permission/not meeting requirements to see the item
+   *
+   * @param inventory the inventory that this item is inside
+   * @param player the player this item is being generated for
+   * @return the generated itemstack, usually used to put in normal bukkit inventories or null if
+   *     the item stack should not be rendered
+   */
+  ItemStack item(InventoryMenu inventory, MatchPlayer player);
```

Comment:
To allow for an item to be "re-rendered", I might suggest the following:
```java
// return false = no render
// return true = item was modified and is ready to go
boolean renderItem(InventoryMenu menu, Player viewer, ItemStack item);
```

Reply:
Based on the current API it's impossible to know if the item has changed without rendering it, which is why invalidation is manual.